src/v/cloud_topics/reconciler/reconciler.cc

```diff
@@ -550,17 +558,23 @@ std::expected<void, reconcile_error> reconciler::add_object_metadata(
     // Remember that there are two kinds of partitions here: the
     // partition of the cloud topic and the partition of the L1 object.
     // There may be multiple partitions in the L1 object for a cloud
-    // topic partition.
+    // topic partition, but right now we only add one per reconciler run.
     for (const auto& commit : obj_meta.commits) {
         auto [first, last] = obj_meta.object_info.index.partitions.equal_range(
           commit.source->topic_id_partition());
+        vassert(
+          std::distance(first, last) == 1,
+          "expected a single partition in the object");
         for (auto it = first; it != last; ++it) {
             const auto& obj_partition = it->second;
             auto add_result = meta_builder->add(
               oid,
               l1::metastore::object_metadata::ntp_metadata{
                 .tidp = commit.source->topic_id_partition(),
-                .base_offset = obj_partition.first_offset,
+                // Use the start offset, this may differ from the base offset in
+                // the metadata info in topics with transactions, as control
+                // batch offsets are skipped.
+                .base_offset = commit.start_offset,
```

Comment:
FWIW it's not just control batches, but aborted transactions will also leave a gap in offset space.